ComponentKit/Utilities/CKComponentDelegateForwarder.mm

```diff
@@ -53,9 +57,51 @@ - (BOOL)respondsToSelector:(SEL)aSelector
   if ([super respondsToSelector:aSelector]) {
     return YES;
   } else {
-    BOOL responds = std::find(_selectors.begin(), _selectors.end(), aSelector) != std::end(_selectors);
-    return responds;
+    return selectorInList(aSelector, _selectors);
+  }
+}
+
+- (void)forwardInvocation:(NSInvocation *)anInvocation
+{
+  // The delegate forwarder is applied as a component view attribute, which is not un-applied to the view on entry
+  // into the reuse pool, yet the *component* property on the view will begin returning nil in this case. This would
+  // turn into a hard-crash because above forwardingTargetForSelector: will return nil, and the method will be directly
+  // invoked on this object. In this case, we have no option but to assert, and not crash, as we do with unhandled
+  // component actions.
+  SEL selector = anInvocation.selector;
+  if (selectorInList(selector, _selectors)) {
+    CKComponent *responder = _view.ck_component;
+    id target = [responder targetForAction:selector withSender:responder];
+    if (!target) {
+      CKFailAssert(@"Delegate method is being called on an unmounted component's view: %@ selector:%@", _view, NSStringFromSelector(selector));
+      return;
+    }
+    [anInvocation invokeWithTarget:target];
+  } else if(selector == @selector(_doNothing)) {
+    
+  } else {
+    [super forwardInvocation:anInvocation];
+  }
+}
+
+- (void)_doNothing {}
+
+- (NSMethodSignature *)methodSignatureForSelector:(SEL)aSelector
+{
+  NSMethodSignature *sig = [super methodSignatureForSelector:aSelector];
+  if (sig) {
+    return sig;
   }
+
+  if (selectorInList(aSelector, _selectors)) {
+    CKComponent *responder = _view.ck_component;
+    id target = [responder targetForAction:aSelector withSender:responder];
+    // We must return a non-nil method signature even if there is no real method signature to return, or we will just
+    // crash in the forwarding system. This ensures the forwardInvocation: call above is called.
+    return (target ? [target methodSignatureForSelector:aSelector] : [self methodSignatureForSelector:@selector(_doNothing)]);
+  }
+
+  return nil;
```

Comment:
Before going down this route I tried a variety of approaches... The standard NSProxy won't work because we actually do need to return a method signature, even if we don't have an alternate target for the selector.